client/components/domains/map-domain-step/index.jsx

```diff
@@ -173,13 +177,20 @@ var MapDomainStep = React.createClass( {
 	},
 
 	addMappingToCart: function( domain ) {
+		this.addPremiumPlanToCart();
```

Comment:
This needs to wrapped in a conditional on `this.props.withPlansOnly`, otherwise we'll always be adding the Premium plan, even for the `original` variation :D